### ISSUE TITLE
Add `best_span_probs` to TransformerQA

### ISSUE THREAD

**Issue by pengshuang** [allenai/allennlp#4465]
Now the calculated logic of ``best_span_scores`` in ``allennlp_models/rc/models/transformer_qa.py`` is:

https://github.com/allenai/allennlp-models/blob/b91333327976e8731bf6866db1308d3d1fc258f2/allennlp_models/rc/models/transformer_qa.py#L136

As the ``best_span_scores`` is the sum of the ``span_start_logits`` and ``span_end_logits`` in the ``best_span``,  **we will get a score that is greater that 1**.

I think that many people would confused about the score (they may think that the ``best_span_scores`` is similar to the probability of the predicted answer, but they will get a score greater than 1).

I suggest to change the calculated logic of best_span_scores as follows (replace the ``span_*_logits`` with ``span_*_probs``):

best_span_scores = (torch.gather(
            span_start_probs, 1, best_spans[:, 0].unsqueeze(1)
        ) + torch.gather(span_end_probs, 1, best_spans[:, 1].unsqueeze(1))) / 2
best_span_scores = best_span_scores.squeeze(1)

Maybe we can also rename the ``best_span_scores`` to ``best_span_probs``.

**Comment by dirkgr** [comment]
For most uses, having a score instead of a probability is good enough, and it comes with extra accuracy in the float32 values to keep it that way. That said, feel free to add another field called `best_span_probs` to the output of `TransformerQA`.

**Comment by ankh6** [comment]
Hi @dirkgr @epwalsh Does someone work on that ? I'd like to make my first contribution to the project

**Comment by epwalsh** [comment]
@ankh6 no one has claimed it so far! So please go ahead if you'd like

**Comment by Abhishek-P** [comment]
Hi Guys, considering this is a "Good First Issue" I will pick it up. 
I have made the changes as mentioned above locally and plan to post the PR in the next few days. 
Please assign the issue to me.

**Comment by dirkgr** [comment]
It doesn't look like @ankh6 is still working on it, so go ahead! Thanks!

**Comment by jacob-morrison** [comment]
https://github.com/allenai/allennlp-models/pull/244 adds span probabilities to TransformerQA's output
